### ISSUE TITLE
date range query

### ISSUE THREAD

**Issue by blaisezzz** [harelba/q#153]
I would like to perform a query to select records in any date range. 

I have downloaded the [single file executable](https://raw.githubusercontent.com/harelba/q/1.6.3/bin/q?source=install_page&table=1) to my local directory as my installation package. The file is renamed to 'q.py'

Then, I generate a log and below is a example of how data looks like in the log file
2017-07-20 15:02:13,512;random logger;CRITICAL;Optimization Event,newvalue: 758, oldvalue: 56, user: Larry, origin: desktop
2017-07-20 15:02:13,512;random logger;INFO;Configuration Event,newvalue: 154, oldvalue: 51, user: Eric, origin: mobile
2017-07-20 15:02:13,512;random logger;WARNING;Control Event,newvalue: 712, oldvalue: 435, user: Saurbh, origin: onlinei
2017-07-20 15:02:13,512;random logger;ERROR;Optimization Event,newvalue: 541, oldvalue: 679, user: Ariel, origin: desktop
2017-07-20 15:02:13,512;random logger;CRITICAL;Status Event,newvalue: 532, oldvalue: 623, user: Badr, origin: online

Next, I execute the python script downloaded like so:
**python "q.py" -t -d ";"  "SELECT * From ./random.log where c1 >= '2017-07-20' between '2017-07-20'"**

Basically, I want to select all the log on the day **2017-07-20**. But, this command doesn't work.

Where's the documentation to perform date range query ? It would be helpful to know.

Thank you.

**Comment by blaisezzz** [comment]
I realized my last comment was missing the proper formatting, here's the updated version:

`
$ cat random.log
2017-07-20 15:02:13,512;random logger;CRITICAL;Optimization Event,newvalue: 758, oldvalue: 56, user: Larry, origin: desktop
2017-07-20 15:02:13,512;random logger;INFO;Configuration Event,newvalue: 154, oldvalue: 51, user: Eric, origin: mobile
2017-07-20 15:02:13,512;random logger;WARNING;Control Event,newvalue: 712, oldvalue: 435, user: Saurbh, origin: onlinei
2017-07-20 15:02:13,512;random logger;ERROR;Optimization Event,newvalue: 541, oldvalue: 679, user: Ariel, origin: desktop
2017-07-20 15:02:13,512;random logger;CRITICAL;Status Event,newvalue: 532, oldvalue: 623, user: Badr, origin: online

$q -t -d ";" "SELECT * From ./random.log where c1 >= '2017-07-20' between '2017-07-20'"

`

**Comment by bitti** [comment]
It still doesn't make sense, since your example doesn't use a range but a single date only. Why not just filter by `where c1 = '2017-07-20'`?

**Comment by stonedbovines** [comment]
If you _do_ actually need ranges then you should refer to a relevant SQL guide, giving something like:
`q "select * from ./random.log where c1 between '2017-07-20' and '2017-07-23'"`
or
`q -d ";" "select * from ./random.log where c1 between '2017-07-20 08:00:00,000' and '2017-07-23 17:51:23,456'"`
